src/doom/m_menu.c

```diff
@@ -1433,6 +1433,23 @@ boolean M_Responder (event_t* ev)
             key = key_menu_activate;
 	    joywait = I_GetTime() + 5;
         }
+
+        // Fill-in the savegame name if user press Fire on the joystick
+        if (saveStringEnter && joybmenu >= 0 && ev->data1&1)
+        {
+            // Create a savegame string
+            char savestring[SAVESTRINGSIZE];
+
+            memset(savestring, 0, SAVESTRINGSIZE);
+            M_snprintf(savestring, SAVESTRINGSIZE - 1, "JOYSTICK SLOT %i", saveSlot);
+            saveCharIndex = strlen(savestring);
+            memcpy(savegamestrings[saveSlot], savestring, SAVESTRINGSIZE);
+
+            // Simulate a KEY_ENTER press. Wait a log time to generate another
+            // keypress.
+            key = KEY_ENTER;
+            joywait = I_GetTime() + 15;
```

Comment:
I was trying to avoid firing the weapon in the game waiting a longer number of ticks but it doesn't work. This is the menu responder and probably the joystick event also gets processed by the game engine. Is there a way to consume the joystick event that generates the slot name and simulates the ENTER key?